### PR TITLE
NCF pipeline refactor (take 2) and initial TPU port.

### PR DESCRIPTION
**Pull request by robieta**
This PR moves data file generation into a side process, replaces lots of serial code with vectorized equivalents, and updates the MLPerf run scripts.

(https://github.com/tensorflow/models/issues/4865) Improved shuffling helps accuracy significantly.
(https://github.com/tensorflow/models/issues/4787) I observed a max memory use of 10.5 GB while testing this branch, so this should address the OOM errors occurring in master.

__CC__
@petermattson 
@reedwm 
@tayo
@codyaustun
@christ1ne

### CODE REVIEW

File: official/recommendation/neumf_model.py [hunk 1]
```diff
@@ -33,102 +33,153 @@
...
+  items = tf.cast(features[movielens.ITEM_COLUMN], tf.int32)
+
+  logits = construct_model(users=users, items=items, params=params)
+
+  if mode == tf.estimator.ModeKeys.PREDICT:
+    predictions = {
+        movielens.RATING_COLUMN: logits,
+    }
+
+    if params["use_tpu"]:
```
Comment (bitfort) [inline official/recommendation/neumf_model.py]: Generally I think we should constants instead of strings... but this isn't that objectionable compared to unblocking our perf teams.
